cherrypy/lib/static.py

```diff
@@ -174,7 +176,22 @@ def _serve_fileobj(fileobj, content_type, content_length, debug=False):
                 cherrypy.log(message, 'TOOLS.STATIC')
             raise cherrypy.HTTPError(416, message)
 
-        if r:
+        is_if_range_in_past = False
```

Comment:
I think that this check would be moved before `httputil.get_ranges()` call, so that we could skip calling another heavy function.
Also, this whole check could be put into its own function (maybe in `httputil`) for readability.